spring-cloud-gcp-data-firestore/src/main/java/org/springframework/cloud/gcp/data/firestore/repository/query/PartTreeFirestoreQuery.java

```diff
@@ -124,9 +130,16 @@ public Object execute(Object[] parameters) {
 			ParameterAccessor paramAccessor = new ParametersParameterAccessor(getQueryMethod().getParameters(),
 					parameters);
 			Pageable pageable = paramAccessor.getPageable();
-			if (pageable != null && pageable.isPaged()) {
-				builder.setOffset((int) Math.min(Integer.MAX_VALUE, pageable.getOffset()));
-				builder.setLimit(Int32Value.newBuilder().setValue(pageable.getPageSize()));
+			if (pageable != null) {
+				if (pageable.isPaged()) {
```

Comment:
Why split into nested `if`?

Reply:
My b, this was an artifact of some experiments we did - reverted.